src/compose/app.ts

```diff
@@ -657,33 +651,22 @@ export class App {
 			return false;
 		}
 
+		// Wait for networks to be created before starting the service
 		if (
 			_.some(
 				networkPairs,
-				(pair) =>
-					`${this.appId}_${pair.target?.name}` === target.config.networkMode,
+				(pair) => pair.target && target.hasNetworkMode(pair.target.name),
 			)
 		) {
 			return false;
 		}
 
+		// Wait for volumes to be created before starting the service
 		if (
-			_.some(target.config.volumes, (volumeDefinition) => {
-				const [sourceName, destName] = volumeDefinition.split(':');
-				if (destName == null) {
-					// If this is not a named volume, ignore it
-					return false;
-				}
-				if (sourceName[0] === '/') {
-					// Absolute paths should also be ignored
-					return false;
-				}
-
-				return _.some(
-					volumePairs,
-					(pair) => `${target.appId}_${pair.target?.name}` === sourceName,
-				);
-			})
+			_.some(
+				volumePairs,
+				(pair) => pair.target && target.hasVolume(pair.target.name),
+			)
```

Comment:
Native Array.prototype.some substitution could be used as in previous comment since volumePairs is also an Array type.